File: LeetCode/0013.go
package leetcode

func romanToInt(s string) int {
	num := map[byte]int{
		'I': 1,
		'V': 5,
		'X': 10,
		'L': 50,
		'C': 100,
		'D': 500,
		'M': 1000,
	}
	ans := 0
	for i := 0; i < len(s)-1; i++ {
		if num[s[i]] >= num[s[i+1]] {
			ans += num[s[i]]
		} else {
			ans -= num[s[i]]
		}
	}
	return ans + num[s[len(s)-1]]
}

File: LeetCode/0013.cpp
class Solution
{
public:
    int num[256];
    int romanToInt(string s)
    {
        num['I'] = 1;
        num['V'] = 5;
        num['X'] = 10;
        num['L'] = 50;
        num['C'] = 100;
        num['D'] = 500;
        num['M'] = 1000;
        long ans = 0;
        for (auto i = 0; i < s.length() - 1; i++)
            if (num[s[i]] >= num[s[i + 1]])
                ans += num[s[i]];
            else
                ans -= num[s[i]];
        return ans += num[s[s.length() - 1]];
    }
};